### PR TITLE
[DOCFIX] Improve documentation on Alluxio configuration

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

File: docs/en/Configuration-Settings.md [hunk 1]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
 
-1. The default values of configuration properties of Alluxio are defined in
-`alluxio-default.properties`. This file can be found in Alluxio source tree and is typically
-distributed with Alluxio binaries. We do not recommend beginner users to edit this file directly.
+## Environment variables
 
-2. Each site deployment and application client can also override the default property values via
-`alluxio-site.properties` file. Note that, this file **must be in the classpath** of the Java VM in
-which Alluxio is running. The easiest way is to copy the site properties template in directory
+There are a few basic but very frequently used Alluxio configuration properties that can be set via the
```
Comment (ifcharming) [inline docs/en/Configuration-Settings.md]: nit: s/but/and ?

File: docs/en/Configuration-Settings.md [hunk 2]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
+remote debugging at port 7001, you can do so using:
+
+{% include Configuration-Settings/more-conf.md %}
+
+One can either set these variables through shell or set them in `conf/alluxio-env.sh`. If this file does not exist yet,
+you can create one from a template we provided in the source code using:
+
+{% include Common-Commands/copy-alluxio-env.md %}
+
+Alternatively, Alluxio can help users bootstrap the `conf/alluxio-env.sh` file by running
```
Comment (ifcharming) [inline docs/en/Configuration-Settings.md]: As @jsimsa mentioned, `bootstrap-conf` should be the recommended way to set up conf. The alternative is setting up alluxio-env manually.

**Comment by ifcharming** [comment]
Thanks @apc999 for the improvement! I left a few comments.

Can you also provide a screenshot of the new page? since there's a new table.

**Comment by jsimsa** [comment]
Running-Alluxio-on-a-Cluster.html needs to be updated as well

**Comment by jsimsa** [comment]
Running-Alluxio-Fault-Tolerant.html needs to be updated

**Comment by jsimsa** [comment]
minor fix is needed for Running-Alluxio-on-EC2-Yarn.html

**Comment by jsimsa** [comment]
minor fix is needed for Running-Hadoop-MapReduce-on-Alluxio.html

**Comment by jsimsa** [comment]
Configuring-Alluxio-with-GCS.html needs to be updated (it talks about JAVA_OPTS in alluxio-env.sh)

**Comment by jsimsa** [comment]
Configuring-Alluxio-with-S3.html needs to be updated in multiple places

**Comment by jsimsa** [comment]
Configuring-Alluxio-with-Swift.html needs to be updated

**Comment by jsimsa** [comment]
minor fix is needed for Configuring-Alluxio-with-GlusterFS.html (removing export)

**Comment by jsimsa** [comment]
minor fix is needed for Configuring-Alluxio-with-HDFS.html (removing export)

**Comment by jsimsa** [comment]
Configuring-Alluxio-with-secure-HDFS.html needs to be updated

**Comment by jsimsa** [comment]
Configuring-Alluxio-with-OSS.html needs to be updated

**Comment by jsimsa** [comment]
minor fix is needed for Configuring-Alluxio-with-NFS.html (removing export)

**Comment by jsimsa** [comment]
took a pass on all the documentation pages and left comments here where I could find outdated references to alluxio-env.sh

File: docs/en/Configuration-Settings.md [hunk 3]
```diff
@@ -8,31 +8,75 @@ priority: 1
 * Table of Contents
 {:toc}
 
-There are two types of configuration parameters for Alluxio:
+This page explains the configuration system of Alluxio and also provides recommendation on how customize the configuration
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: `... how to customize ...`

File: docs/en/Configuration-Settings.md [hunk 4]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
 
-2. Each site deployment and application client can also override the default property values via
-`alluxio-site.properties` file. Note that, this file **must be in the classpath** of the Java VM in
-which Alluxio is running. The easiest way is to copy the site properties template in directory
+There are a few basic but very frequently used Alluxio configuration properties that can be set via the
+following environment variables:
+
+<table class="table table-striped">
+<tr><th>Environment Variable</th><th>Meaning</th></tr>
+<tr><td>`$ALLUXIO_MASTER_HOSTNAME`</td><td>hostname of Alluxio master, default to localhost.</td></tr>
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: s/default/defaults/

File: docs/en/Configuration-Settings.md [hunk 5]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
+<tr><td>`$ALLUXIO_MASTER_ADDRESS`</td><td>Same as `ALLUXIO_MASTER_HOSTNAME`, deprecated since version 1.1 and
+will be remove in version 2.0.</td></tr>
+<tr><td>`$ALLUXIO_UNDERFS_ADDRESS`</td><td>under storage system address, default to
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: s/default/defaults

File: docs/en/Configuration-Settings.md (hunk 4 above)
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: saying "default to ..." is not grammatically correct

either say "defaults to ..." or "using ... as the default"

Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: this applies to all the table entries

File: docs/en/Configuration-Settings.md [hunk 6]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
-2. Each site deployment and application client can also override the default property values via
-`alluxio-site.properties` file. Note that, this file **must be in the classpath** of the Java VM in
-which Alluxio is running. The easiest way is to copy the site properties template in directory
+There are a few basic but very frequently used Alluxio configuration properties that can be set via the
+following environment variables:
+
+<table class="table table-striped">
+<tr><th>Environment Variable</th><th>Meaning</th></tr>
+<tr><td>`$ALLUXIO_MASTER_HOSTNAME`</td><td>hostname of Alluxio master, default to localhost.</td></tr>
+<tr><td>`$ALLUXIO_MASTER_ADDRESS`</td><td>Same as `ALLUXIO_MASTER_HOSTNAME`, deprecated since version 1.1 and
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: use consistent capitalization for all the description, either capitalize all or none (unless it is a proper name such as Java)

File: docs/en/Configuration-Settings.md [hunk 7]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
+`${ALLUXIO_HOME}/underFSStorage` which is a local file system.</td></tr>
+<tr><td>`$ALLUXIO_RAM_FOLDER`</td><td>The directory where a worker stores in-memory data, default to /mnt/ramdisk.</td></tr>
+<tr><td>`$ALLUXIO_JAVA_OPTS`</td><td>Java VM options for both Master and Worker.</td></tr>
+<tr><td>`$ALLUXIO_MASTER_JAVA_OPTS`</td><td>additional Java VM options for Master configuration.</td></tr>
+<tr><td>`$ALLUXIO_WORKER_JAVA_OPTS`</td><td>additional Java VM options for Worker configuration. Note that, by
+default `ALLUXIO_JAVA_OPTS` is included in both `ALLUXIO_MASTER_JAVA_OPTS` and
+`ALLUXIO_WORKER_JAVA_OPTS`.</td></tr>
+</table>
+
+For example, if you would like to setup a Alluxio master at `localhost` that talks to a HDFS also running at `localhost`, and enable Java
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: "... an Alluxio master ... an HDFS cluster ..."

File: docs/en/Configuration-Settings.md (hunk 2 above)
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: yes, we should guide users towards using the bootstrap-conf script over copying the template manually

File: docs/en/Configuration-Settings.md [hunk 8]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
+{% include Common-Commands/copy-alluxio-env.md %}
+
+Alternatively, Alluxio can help users bootstrap the `conf/alluxio-env.sh` file by running
+
+{% include Common-Commands/bootstrap-conf.md %}
+
+
+## Configuration properties
+
+In addition to these environment variables that provide very basic setting, Alluxio also provides a
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: remove "very"

File: docs/en/Configuration-Settings.md [hunk 9]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
+more general approach for users to customize all supported configuration properties via property files.
+On startup, Alluxio optionally loads configuration properties file to set the configuration properties:
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: "On startup, Alluxio checks if certain files exist if so, it uses their content to override the default values of configuration properties. In particular:"

File: docs/en/Configuration-Settings.md [hunk 10]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
+2. Alluxio master and workers will load `alluxio-server.properties` file; Alluxio clients (e.g., Jobs reading from or writing to
+Alluxio) will load `alluxio-client.properties` file.
+
+Note that, these property files are searched in `${HOME}/.alluxio/`, `/etc/conf/` and the classpath of the Java VM in
+which Alluxio is running in order. The easiest way is to copy the site properties template in directory
 `$ALLUXIO_HOME/conf` and edit it to fit your configuration tuning needs.
 
 ```bash
-$ cp $ALLUXIO_HOME/conf/alluxio-site.properties.template $ALLUXIO_HOME/conf/alluxio-site.properties
+$ cp $ALLUXIO_HOME/conf/alluxio-site.properties.template ~/.alluxio/alluxio-site.properties
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: should this go to Common-Commands?

File: docs/en/Configuration-Settings.md [hunk 11]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
+
+
+## Configuration properties
+
+In addition to these environment variables that provide very basic setting, Alluxio also provides a
+more general approach for users to customize all supported configuration properties via property files.
+On startup, Alluxio optionally loads configuration properties file to set the configuration properties:
+
+1. For each Alluxio site deployment, both servers or application clients can override the default property values via
+`alluxio-site.properties` file.
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: what locations are checked for this file?

File: docs/en/Configuration-Settings.md [hunk 12]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
+
+2. Alluxio master and workers will load `alluxio-server.properties` file; Alluxio clients (e.g., Jobs reading from or writing to
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: Alluxio master and workers will load the `alluxio-server.properties` file, while Alluxio clients, such as jobs reading from or writing to Alluxio, will load the `alluxio-client.properties` file

File: docs/en/Configuration-Settings.md [hunk 13]
```diff
@@ -8,31 +8,75 @@ priority: 1
...
+for Alluxio users in different contexts.
 
-1. [Configuration properties](#configuration-properties) are used to configure the runtime settings
-of Alluxio system, and
-2. [System environment properties](#system-environment-properties) control the Java VM options to
-run Alluxio as well as some basic very setting.
+# Configuration in Alluxio
 
-# Configuration properties
+Alluxio runtime respects two sources of configuration settings:
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: this page should comment on the priority of configuration settings? defaults < properties files < environment variables

**Comment by jsimsa** [comment]
@apc999 did a pass on your configuration changes too

File: docs/en/Configuration-Settings.md (hunk 3 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: ack

File: docs/en/Configuration-Settings.md (hunk 13 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: added

File: docs/en/Configuration-Settings.md (hunk 1 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

File: docs/en/Configuration-Settings.md (hunk 4 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: thanks for the suggestion. they have been fixed.

File: docs/en/Configuration-Settings.md (hunk 6 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

File: docs/en/Configuration-Settings.md (hunk 7 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

File: docs/en/Configuration-Settings.md (hunk 2 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

File: docs/en/Configuration-Settings.md (hunk 8 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

File: docs/en/Configuration-Settings.md (hunk 9 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

File: docs/en/Configuration-Settings.md (hunk 12 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

File: docs/en/Configuration-Settings.md (hunk 10 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

**Comment by jsimsa** [comment]
@apc999 did you push your changes yet?

**Comment by apc999** [comment]
Running-Alluxio-on-a-Cluster.html needs to be updated as well 
-> done
Running-Alluxio-Fault-Tolerant.html needs to be updated
-> done
minor fix is needed for Running-Alluxio-on-EC2-Yarn.html (removing export)
-> done
minor fix is needed for Running-Hadoop-MapReduce-on-Alluxio.html (removing export)
-> done
Configuring-Alluxio-with-GCS.html needs to be updated (it talks about JAVA_OPTS in alluxio-env.sh)
=> done
Configuring-Alluxio-with-S3.html needs to be updated in multiple places
=> done
Configuring-Alluxio-with-Swift.html needs to be updated
=> done
minor fix is needed for Configuring-Alluxio-with-GlusterFS.html (removing export)
=> done
minor fix is needed for Configuring-Alluxio-with-HDFS.html (removing export)
=>done
minor fix is needed for Configuring-Alluxio-with-secure-HDFS.html (removing export)
=>done
Configuring-Alluxio-with-OSS.html needs to be updated
=>done
minor fix is needed for Configuring-Alluxio-with-NFS.html (removing export)
=>done

**Comment by apc999** [comment]
@jsimsa @ifcharming I fixed a bunch of documentation on configuration. PTAL

File: docs/_includes/Configuring-Alluxio-with-GCS/google.md [hunk 14]
```diff
@@ -1,2 +1,6 @@
-    -Dfs.gcs.accessKeyId=<GCS_ACCESS_KEY_ID>
+```bash
+ALLUXIO_JAVA_OPTS="
+    -Dfs.gcs.accessKeyId=<GCS_ACCESS_KEY_ID> 
     -Dfs.gcs.secretAccessKey=<GCS_SECRET_ACCESS_KEY>
+"
```
Comment (calvinjia) [inline docs/_includes/Configuring-Alluxio-with-GCS/google.md]: Do we want `+=` for consistency?

Comment (apc999) [inline docs/_includes/Configuring-Alluxio-with-GCS/google.md]: done

**Comment by apc999** [comment]
@calvinjia  PTAL. thanks!

**Comment by haoyuan** [comment]
+1

**Pull request merged**